### ISSUE TITLE
406 Not Acceptable

### ISSUE THREAD

**Issue by tsteuwer** [digitalsadhu/loopback-component-jsonapi#31]
It looks like we are not doing something correctly. The JSON API spec is shows:

> Servers **MUST** respond with a `406 Not Acceptable` status code if a request's `Accept` header contains the JSON API media type and all instances of that media type are modified with media type parameters.

However, when sending an actual `POST` message with `Accept: application/vnd.api+json`, the component is responding with a `406 Not Acceptable`.

**Comment by digitalsadhu** [comment]
I'm not quite following. So you send a post setting the Accept header to application/vnd.api+json and you get back a 406 Not Acceptable but it should only be doing that if "all instances of that media type are modified with the media type parameters"?

**Comment by tsteuwer** [comment]
I'm not exactly sure whats going on with the backend, but it's not accepting an actual `POST` request. Let me show you how.

**Request Header**
```
POST /api/people HTTP/1.1
Host: 127.0.0.1:3000
Connection: keep-alive
Content-Length: 170
Cache-Control: max-age=0
Accept: application/vnd.api+json
Origin: http://127.0.0.1:4200
CSP: active
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.132 Safari/537.36
Content-Type: application/vnd.api+json
Referer: http://127.0.0.1:4200/
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.8

```

Fine and dandy, just creating a new `Person` model using `POST`. However, the API is responding with:

**Response Headers**
```
HTTP/1.1 406 Not Acceptable
X-Powered-By: Express
Access-Control-Allow-Origin: http://127.0.0.1:4200
Vary: Origin, Accept-Encoding
Access-Control-Allow-Credentials: true
Content-Type: application/vnd.api+json; charset=utf-8
Location: http://127.0.0.1:3000/api/people/2
Content-Length: 14
ETag: W/"e-wYgSOoR59EeG6OGsAfOFAA"
Date: Fri, 23 Oct 2015 21:18:02 GMT
Connection: keep-alive
```

Here's the full cURL request:
```
curl 'http://127.0.0.1:3000/api/people' -H 'Origin: http://127.0.0.1:4200' -H 'Accept-Encoding: gzip, deflate' -H 'Accept-Language: en-US,en;q=0.8' -H 'CSP: active' -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.132 Safari/537.36' -H 'Content-Type: application/vnd.api+json' -H 'Accept: application/vnd.api+json' -H 'Cache-Control: max-age=0' -H 'Referer: http://127.0.0.1:4200/' -H 'Connection: keep-alive' --data-binary '{"data":{"attributes":{"firstName":"Joe","lastName":"Shmoe","age":31,"createdAt":"2015-10-23T21:18:02.416Z","lastModifiedAt":"2015-10-23T21:18:02.416Z"},"type":"people"}}' --compressed
```

And, here's the model `JSON` file for loopback if you want to try it yourself.
```
{
  "name": "people",
  "base": "PersistedModel",
  "idInjection": true,
  "options": {
    "validateUpsert": true
  },
  "properties": {
    "firstName": {
      "type": "string",
      "required": true
    },
    "lastName": {
      "type": "string",
      "required": true
    },
    "age": {
      "type": "number",
      "required": true
    },
    "createdAt": {
      "type": "date"
    },
    "lastModifiedAt": {
      "type": "date"
    }
  },
  "validations": [],
  "relations": {},
  "acls": [],
  "methods": {}
}
```

And here's the javascript file:
```
module.exports = function(People) {

};
```

**Comment by tsteuwer** [comment]
I found the issue! You guys do not accept `application/vnd.api+json` as an `Accept` header! Need to add this :)

**Comment by digitalsadhu** [comment]
Awesome! Should be an easy fix! Will add a failing test and fix tonight.

Thanks for looking into this, great stuff!

**Comment by digitalsadhu** [comment]
Fixed in release v0.5.0
Published to npm
@tsteuwer

**Comment by tsteuwer** [comment]
Awesome @digitalsadhu!!
